### PR TITLE
Wire up modular mode for outputting to the render pipeline.

### PR DESCRIPTION
**Pull request by veluca93**
Also wire up DC upscaling.

### CODE REVIEW

**Comment by jonsneyers** [comment]
Does this work also for alpha/extra channels, or is this step just making it work for the no-VarDCT RGB case?

**Comment by veluca93** [comment]
> Does this work also for alpha/extra channels, or is this step just making it work for the no-VarDCT RGB case?

Extra channels coming up soon :)

File: lib/jxl/dec_modular.cc [hunk 1]
```diff
@@ -622,8 +636,34 @@ Status ModularFrameDecoder::FinalizeDecoding(PassesDecoderState* dec_state,
 
   auto& decoded = dec_state->decoded;
 
-  JXL_RETURN_IF_ERROR(
-      ModularImageToDecodedRect(gi, dec_state, pool, output, Rect(decoded)));
+  if (dec_state->render_pipeline) {
+    std::atomic<size_t> num_errors{0};
```
Comment (sboukortt) [inline lib/jxl/dec_modular.cc:640]: [In the past](https://github.com/libjxl/libjxl/pull/860/files#diff-c7ffee7b87f9e62e559021b02af4ddbeae7d920d0ac7bde1a3b8322e4a9319adR45), I have used a bool for this. Is a size_t preferable? It does maybe seem a bit lighter than the bool notation-wise.

**Review by sboukortt — DISMISSED** [review]

**Review by jonsneyers — APPROVED** [review]

File: lib/jxl/dec_modular.cc (hunk 1 above)
Comment (veluca93) [reply lib/jxl/dec_modular.cc]: Done.

**Comment by veluca93** [comment]
Had to modify the code to deal with GetInputBuffers() multiple times, PTAL.

**Review by jonsneyers — APPROVED** [review]

**Pull request merged**